web-api-db-ctrl/postgres/tenant/20230713173212_document.sql

```diff
@@ -0,0 +1,29 @@
+-- Copyright 2023 Xayn AG
+--
+-- This program is free software: you can redistribute it and/or modify
+-- it under the terms of the GNU Affero General Public License as
+-- published by the Free Software Foundation, version 3.
+--
+-- This program is distributed in the hope that it will be useful,
+-- but WITHOUT ANY WARRANTY; without even the implied warranty of
+-- MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+-- GNU Affero General Public License for more details.
+--
+-- You should have received a copy of the GNU Affero General Public License
+-- along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+CREATE TABLE snippet (
+    document_id TEXT NOT NULL
+        REFERENCES document(document_id) ON DELETE CASCADE,
+    snippet_idx INTEGER NOT NULL,
+    snippet TEXT NOT NULL,
+    embedding FLOAT4[] NOT NULL,
+
+    PRIMARY KEY (document_id, snippet_idx)
+);
+
+INSERT INTO snippet(document_id, snippet_idx, snippet, embedding)
+    SELECT document_id, 0, snippet, embedding FROM document;
+
+ALTER TABLE document RENAME COLUMN snippet TO raw_document;
```

Comment:
We have to properly decide how the handle "raw documents" (which long term could be e.g. PDF), especially wrt. upsertion, applied size limits and billing.